lib/npm.js

```diff
@@ -77,7 +77,26 @@ module.exports.install = function (packagesToUpdate, options) {
             if (!err) {
                 if (options.save) {
                     try {
-                        fs.writeFileSync(packagePath, fs.readFileSync(packagePath, 'utf8').replace(/[~^]/g, ''));
+                        var packageJSON = JSON.parse(fs.readFileSync(packagePath, 'utf8'));
+                        if (options.dev) {
```

Comment:
What about 

```javascript
var dependencyType = options.dev ? 'devDependencies' : 'dependencies';
```

then

```javascript
packageJSON.devDependencies[packName] = packageJSON ...
```

instead of the `if` statement?